olive/model/config/hf_config.py

```diff
@@ -245,3 +246,11 @@ def task_or_model_class_required(cls, v, values):
             if not v and not values.get("task", None):
                 raise ValueError("Either task or model_class must be specified")
         return v
+
+
+def get_loading_args_from_pretrained(hf_config: HfConfig) -> Dict[str, Any]:
+    return hf_config.from_pretrained_args.get_loading_args() if hf_config.from_pretrained_args else {}
+
+
+def get_model_type(hf_config: HfConfig) -> str:
+    return get_hf_model_config(hf_config.model_name, **get_loading_args_from_pretrained(hf_config)).model_type
```

Comment:
I suppose @jambayk means to move this function into hf_utils.py with the following signature. 

def get_model_type(model_name: str, **kwargs):

Reply:
anyway, you can keep it as is. it is not big deal.

Reply:
the major difference between this one with https://github.com/microsoft/Olive/blob/ab0af2333e17e06d2245640f1759d1b7eed955f5/olive/model/config/io_config.py#L84 @jambayk mentioned is:
the get_model_type need call transformers. that one @jambayk point to need not. 

That's the reason we suggest to move it into hf_utils.

Reply:
the get_model_type only accept the model name and loading arguments, which need to be called in conversion.py
like:
```python
        hf_config = HfConfig(**model_config["hf_config"])
        model_type = get_model_type(hf_config.model_name, **hf_config.get_from_pretrained_args())
```

Reply:
Resolving with introducing hf_utils dependency in hf_config